### PR TITLE
Update of markers when editing text

### PR DESCRIPTION
**Pull request by SergioEstevao**
Refs #146 and #147 

This PR implement support for updating/removing lists while editing text inside the list, specially when deleting around markers and entering new lines.

How to test:

- Start the editor demo
- Create a new ordered list spreading across several lines
- Try the following:
  - Break lines in middle and see if the markers are update correctly
  - Try Delete marker using the delete key or cutting a selection and see if markers are updated correctly
 - Enter two new empty lines in the end of the list and see if the list is interrupted.
  - Enter two new empty new lines in a go in the middle of a list and see if the list is split int half

### CODE REVIEW

File: Aztec/Classes/Formatters/TextListFormatter.swift [hunk 1]
```diff
@@ -37,6 +37,39 @@ class TextListFormatter
         let listParagraphs = string.paragraphRanges(spanningRange: listRange)
         return applyList(ofStyle: style, toString: string, atRanges: listParagraphs)
     }
+
+    func refreshList(inString string: NSMutableAttributedString, atRange range: NSRange) -> NSRange? {
```
Comment (diegoreymendez) [inline Aztec/Classes/Formatters/TextListFormatter.swift]: I think this method requires some documentation explaining:

- Why does this method need to refresh the list?
- How does the refresh work?

File: Aztec/Classes/Formatters/TextListFormatter.swift [hunk 2]
```diff
@@ -37,6 +37,39 @@ class TextListFormatter
...
+        }
+        if let textList = string.attribute(TextList.attributeName, atIndex: firstParagraphRange.location, effectiveRange: nil) as? TextList {
+            styleOptional = textList.style
+        }
+
+        guard let style = styleOptional else {
+            return removeList(fromString: string, atRanges: [firstParagraphRange])
+        }
+
+
```
Comment (diegoreymendez) [inline Aztec/Classes/Formatters/TextListFormatter.swift]: Nitpicky: Remove 1 line break.

File: Aztec/Classes/TextKit/TextView.swift [hunk 3]
```diff
@@ -455,6 +496,37 @@ public class TextView: UITextView {
         selectedRange = NSRange(location:selectionStartRange.location, length: selectionEndRange.location - selectionStartRange.location)
     }
 
+    // MARK: - List Code
+
+    private func refreshListsIfOnePrecedes(range range:NSRange) {
```
Comment (diegoreymendez) [inline Aztec/Classes/TextKit/TextView.swift]: Definitely needs documentation - kinda wondering what "if one precedes" means.

Comment (SergioEstevao) [inline Aztec/Classes/TextKit/TextView.swift]: @diegoreymendez ready for another look.

**Pull request merged**